bukkit/src/main/java/me/lucko/luckperms/bukkit/listeners/BukkitConnectionListener.java

```diff
@@ -240,7 +240,7 @@ public void onPlayerQuit(PlayerQuitEvent e) {
 
         // perform unhooking from bukkit objects 1 tick later.
         // this allows plugins listening after us on MONITOR to still have intact permissions data
-        this.plugin.getBootstrap().getServer().getScheduler().runTaskLater(this.plugin.getLoader(), () -> {
+        this.plugin.getBootstrap().getScheduler().sync(player, () -> {
```

Comment:
This part here currently does not seem to work on folia, since the player's scheduler gets retired and the task isn't ran.